src/main/java/org/eclipse/yasson/internal/JsonbConfigProperties.java

```diff
@@ -156,25 +156,18 @@ private String getGlobalConfigJsonbDateFormat() {
         }).orElse(JsonbDateFormat.DEFAULT_FORMAT);
     }
 
-    private PropOrderStrategy initOrderStrategy() {
-        Optional<String> strategy = getPropertyOrderStrategy();
-        if (strategy.isPresent()) {
-            switch (strategy.get()) {
-                case PropertyOrderStrategy.LEXICOGRAPHICAL:
-                    return new LexicographicalOrderStrategy();
-                case PropertyOrderStrategy.REVERSE:
-                    return new ReverseOrderStrategy();
-                case PropertyOrderStrategy.ANY:
-                    return new AnyOrderStrategy();
-                default:
-                    throw new JsonbException(Messages.getMessage(MessageKeys.PROPERTY_ORDER, strategy));
-            }
+    private Function<Collection<PropertyModel>, List<PropertyModel>> initOrderStrategy() {
+        String strategy = getPropertyOrderStrategy();
+        
+        if(strategy == null) {
+            //default by spec
+            return StrategiesProvider.getOrderingFunction(PropertyOrderStrategy.LEXICOGRAPHICAL);
         }
-        //default by spec
-        return new LexicographicalOrderStrategy();
+        
+        return StrategiesProvider.getOrderingFunction(strategy);
     }
 
-    private Optional<String> getPropertyOrderStrategy() {
+    private String getPropertyOrderStrategy() {
```

Comment:
I have to say that I don't see the reason why to remove Optional here.

Reply:
I guess that makes sense, keep it unified with most of the other methods.